lib/app/socket-handler.js

```diff
@@ -1,31 +1,43 @@
 'use strict';
 
-var crawler = require('./crawler.js');
-var intervalProduction;
-var userCount = 0;
-
-
 module.exports = function socketHandler(config, io) {
+  var crawler = require('./crawler.js');
+  var intervals = {
+    hydraters: null,
+    providers: null
+  };
+  var userCount = {
+    hydraters: 0,
+    providers: 0
+  };
+
   io.on('connection', function(socket) {
-    socket.emit('urls', config.urls);
-    userCount += 1;
+    // source can either be 'providers' or 'hydraters'
+    socket.on('source', function(source) {
+      socket.join(source);
+      socket.emit('urls', config.urls[source]);
 
-    if(userCount !== 1) {
-      crawler.sendSavedData(socket);
-    }
+      userCount[source] += 1;
 
-    if(!intervalProduction) {
-      intervalProduction = setInterval(function() {
-        crawler.pollData(socket, config.urls, config.timeout);
-      }, config.pollingInterval);
-    }
+      // send cached data
+      if(userCount[source] !== 1) {
+        crawler.sendSavedData(socket, source);
+      }
 
-    socket.on('disconnect', function() {
-      userCount -= 1;
-      if(userCount === 0) {
-        clearInterval(intervalProduction);
-        intervalProduction = null;
+      if(!intervals[source]) {
+        intervals[source] = setInterval(function() {
+          crawler.pollData(socket, config.urls[source], config.timeout, source);
+        }, config.pollingInterval);
       }
+
+      // stop polling on disconect if they are no more users
+      socket.on('disconnect', function() {
+        userCount[source] -= 1;
+        if(userCount[source] === 0) {
```

Comment:
I'm pretty sure there is absolutely no way the userCount could be < 0, but yeah, just in case.